### DISCUSSION TITLE
How do I get value hints in IntelliJ IDE?

### DISCUSSION THREAD

**Discussion by ghost** [kcl-lang/kcl#1657] [Q&A] [answered]
I installed KCL language server extension, this works fairly well for basic type hints, but for defined values, I get nothing.

I've used pkl before and values defined in a config were available in the type hinting.
example:
```
schema App:
    namespace: "space1" | "space2"
```

when defining the namespace for app, I would expect to see a type hint showing the available values.

**Comment by Peefy** [comment]
![image](https://github.com/user-attachments/assets/1eec6b5e-bcf4-496d-9921-3858c01f5f69)

Hello, do you mean the values of `"space1"` and `"space2"` when enter the `:` char or enter the `"` or `s` chars?

cc @He1pa

**Reply by ghost** [reply]
This is exactly what I mean. Do I specify this wrong? I followed the abstraction example to generate the kubernetes yaml file for my projects. My app contains all schemas and I have a build.k file which renders the yaml, just like the kubernetes_render.k in the abstraction example.

```
import app
_name = "App name"

app.App {
    name = _name
    namespace = "space1"
}
```

**Comment by He1pa** [comment]
Do you means want to display attribute type hint when defining config? e.g.
```
schema App:
    namespace: "space1" | "space2"

app [:APP] = App{
    namespace [: "space1" | "space2"] : "space1"
}
```
Where [:APP] and `[: "space1" | "space2"]` is hint

**Reply by ghost** [reply]
That is correct, when typing the : or quote or press ctrl + space for auto complete, I would assume that I get the available values as hint.

**Comment by He1pa** [comment]
I have impl this feature in https://github.com/kcl-lang/kcl/pull/1659.It will be available in the next version

**Comment by He1pa** [comment, marked as answer]
Currently, there are several situations where prompt information is provided.
1, when typing config attr name, provide complete item and it type
![image](https://github.com/user-attachments/assets/05d5a66e-8dd0-4867-98b0-d453bea39b9a)

2. after this, provide type hint for attr
![image](https://github.com/user-attachments/assets/78ae1f95-a733-4fb4-857e-5b3e47ef70bb)

3. hover info
![image](https://github.com/user-attachments/assets/6acc409f-971f-45a5-9c25-2fbd4e93d1d5)
 
4. when typing `:` or `=`，provide available complete item according  by it type
![image](https://github.com/user-attachments/assets/0c7bacaa-405b-48ac-9bdc-7d405e3e23aa)
bool -> True and False
str -> ""
list -> []
dict -> {}
Union -> each item
schema -> pkg.schemaName {}

If you have more feat, welcome for your feedback

**Comment by ghost** [comment]
I was waiting for this in the new release, but just saw that is has been reverted. Why? And can we expect that is will be in the next release? This is really a show stopper for us as a company to use KCL.

**Comment by Peefy** [comment]
cc @He1pa

**Comment by He1pa** [comment]
We have encountered some problems with the display of hints on some expressions. And it is difficult to solve in a short time. After the problem is solved, we will add this feature again.

**Comment by He1pa** [comment]
We re-provide this feature in this PR: https://github.com/kcl-lang/kcl/pull/1751
